chat-widget/src/components/livechatwidget/common/ActivitySubscriber/AddActivitySubscriber.ts

```diff
@@ -0,0 +1,72 @@
+import ChatWidgetEvents from "../ChatWidgetEvents";
+import { IActivitySubscriber } from "./IActivitySubscriber";
+
+/**
+ * The `AddActivitySubscriber` class is responsible for subscribing to the `ADD_ACTIVITY` event
+ * from the `ChatWidgetEvents` and notifying an observer when a new activity is added.
+ * 
+ * This class implements the `IActivitySubscriber` interface and acts as a bridge between
+ * the event system and the observer pattern.
+ */
+export class AddActivitySubscriber implements IActivitySubscriber {
+    /**
+     * The observer that will be notified when a new activity is added.
+     * This is expected to be an object with a `next` method, such as an RxJS `Observer`.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public observer: any;
+    
+    /**
+     * Set to track processed activity IDs to prevent duplicate processing.
+     */
+    private processedActivityIds: Set<string> = new Set();
+    
+    /**
+     * Constructor initializes the `AddActivitySubscriber` and sets up an event listener
+     * for the `ChatWidgetEvents.ADD_ACTIVITY` event. When the event is triggered, it checks
+     * if the event payload contains an `activity` and notifies the observer.
+     */
+    constructor() {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        window.addEventListener(ChatWidgetEvents.ADD_ACTIVITY, (event: any) => {
+            // Check if the event contains the expected payload and activity
+            if (event?.detail?.payload?.activity) {
+                const activity = event.detail.payload.activity;
+                const activityId = activity.id;
+
+                if (activity.identifier) {
+                    if (this.processedActivityIds.has(activity?.identifier)) {
+                        return; // Skip processing if already handled
+                    }
+                    // Add the activity ID to the processed set
+                    this.processedActivityIds.add(activity?.identifier);
```

Comment:
Do we have a cleanup logic to avoid memory leak?

Reply:
not sure what you mean, this only push objects to current stream of activities,